TheWarWithin/RogueAssassination.lua

```diff
@@ -2434,27 +2432,36 @@ spec:RegisterAbilities( {
                 debuff.rupture_deathmark.pmultiplier = persistent_multiplier
             end
 
-            if buff.indiscriminate_carnage_any.up then
-                active_dot.rupture = min( true_active_enemies, active_dot.rupture + 2 )
-            end
-
-            if buff.finality_rupture.up then removeBuff( "finality_rupture" )
-            elseif talent.finality.enabled then applyBuff( "finality_rupture" ) end
-
+            spend( combo_points.current, "combo_points" )
+            if talent.supercharger.enabled then removeStack( "supercharged_combo_points" ) end
 
+            --- Assassination Rogue specific
 
             if talent.scent_of_blood.enabled or azerite.scent_of_blood.enabled then
                 applyBuff( "scent_of_blood", dot.rupture.remains, active_dot.rupture )
             end
 
-            spend( combo_points.current, "combo_points" )
-            removeStack( "supercharged_combo_points" )
+            if buff.indiscriminate_carnage_any.up then
+                active_dot.rupture = min( true_active_enemies, active_dot.rupture + 2 )
+            end
 
             if buff.serrated_bone_spike_charges.up then
-                gain ( 1 + buff.serrated_bone_spike_charges.stack, "combo_points" )
-                removeStack( "serrated_bone_spike_charges" )
+                for i = 1, buff.indiscriminate_carnage_any.up and 3 or 1 do
```

Comment:
`gain` is a little bit computationally expensive since it triggers a new forecast of resources each time.  Can this be done without the loop?  Also, do we need to include a check for `true_active_enemies` here?  If there's only 1 or 2 targets, will the Serrated Bone Spike charges get consumed 3 times?

`removeStack` can take a second argument to remove multiple stacks if we know how many to do in total.

For 1, 2, and 3 targets and SBS stacks and Indiscriminate Carnage up:
* How many Combo Points are gained?
* How many stacks of SBS are consumed?

Can we calculate these numbers and fire `gain` and `removeStack` once?

Reply:
Created a function to handle all the BOnespike BS, which only calls the gain one time at the end.